osbenchmark/client.py

```diff
@@ -278,3 +280,124 @@ async def create(params):
             return await KafkaMessageProducer.create(params)
         else:
             raise ValueError(f"Unsupported ingestion source type: {producer_type}")
+
+
+class GrpcClientFactory:
+    """
+    Factory for creating gRPC client stubs.
+    Note gRPC channels must default `use_local_subchannel_pool` to true.
+    Sub channels manage the underlying connection with the server. When the global sub channel pool is used gRPC will
+    re-use sub channels and their underlying connections which does not appropriately reflect a multi client scenario.
```

Comment:
Probably could be clearer: using local subchannels permits additional connections to be created each with their own pools, which can improve performance.